api/envoy/extensions/clusters/dynamic_forward_proxy/v3/cluster.proto

```diff
@@ -42,6 +42,12 @@ message ClusterConfig {
   // bar.example.com if that also resolved to 1.2.3.4.
   //
   // .. note::
+  //   By design, this feature will maximize reuse of connections. This means that instead
+  //   opening a new connection when an existing connection reaches the maximum number of
```

Comment:
Well, with just the code in this PR none of the clusters actually implement selectPool() so nothing will happen :) (More PRs forthcoming). Once Alyssa's adds request queueing and all the other PRs land max_concurrent_streams will be honored. It will limit the number of concurrent streams set on the connection. However, requests will be queued before going to the wire if there are no available streams. (Pending details in Alyssa's to be written PR)

Reply:
Got it, thanks for the explanation!